app/screens/receive-bitcoin-screen/receive-usd.tsx

```diff
@@ -145,6 +146,45 @@ const ReceiveUsd = () => {
   const { lnUpdate } = useSubscriptionUpdates()
   const [showMemoInput, setShowMemoInput] = useState(false)
   const [showAmountInput, setShowAmountInput] = useState(false)
+  const { timeLeft, startCountdownTimer, resetCountdownTimer } = useCountdownTimer()
+  const timeUntilInvoiceExpires = useMemo(() => {
+    if (invoice) {
+      return (
+        getLightningInvoiceExpiryTime(decodeInvoiceString(invoice.paymentRequest)) -
+        Math.round(Date.now() / 1000)
+      )
+    }
+    return NaN
+  }, [invoice])
+
+  useEffect(() => {
```

Comment:
You don't have to make this change, but one way to avoid having to keep the timer in sync like this would be to implement the hook by storing the expiration of the timer, and then each second calculating how much time is left, vs the current implementation where you store how much time is left but then it can get out of sync if you leave the app.

Reply:
I made your suggested change.